packages/overlay/src/utils/instrumentation.ts

```diff
@@ -0,0 +1,62 @@
+import * as Sentry from '@sentry/react';
+import { createRoutesFromChildren, matchRoutes, useLocation, useNavigationType } from 'react-router-dom';
+import { useEffect } from '../react-instance';
+
+export default function initSentry(initialTab: string, options: Sentry.BrowserOptions = {}) {
+  const sentryTraceParent: Record<string, string> = {};
+  const navTiming = performance.getEntriesByType('navigation');
+  if (navTiming.length > 0) {
+    const serverTiming = (navTiming[0] as PerformanceNavigationTiming).serverTiming;
+    if (serverTiming && serverTiming.length > 0) {
+      for (const { name, description } of serverTiming) {
+        if (name === 'sentryTrace' || name === 'baggage') {
+          sentryTraceParent[name] = description;
+        }
+      }
+    }
+  }
+  const hasTraceParent = Object.keys(sentryTraceParent).length === 2;
+  const integrations = [
+    // See docs for support of different versions of variation of react router
+    // https://docs.sentry.io/platforms/javascript/guides/react/configuration/integrations/react-router/
+    Sentry.reactRouterV6BrowserTracingIntegration({
```

Comment:
h: how would we keep the traces created from this integration isolated from the trace on the page? Have you tested if they interfere?

Reply:
This is not relevant as we only enable Sentry in the overlay when it is loaded from the Sidecar as a full-blown app in full page mode.